### ISSUE TITLE
VaultDynamicSecret with static ldap secret is not refreshing password.

### ISSUE THREAD

**Issue by ninjadude333** [hashicorp/vault-secrets-operator#225]
**Describe the bug**
When using VaultDynamicSecret with a static ldap secret (that is configured to auto rotate password every 6 hours)
the new change is not detected by the VaultDynamicSecrets, and there is not option to add a refresh interval like in VaultStaticSecret

**To Reproduce**
1. create ldap static role:
```
vault write ldap/static-role/hashicorp \
    dn='uid=hashicorp,ou=People,dc=ninjadude,dc=com' \
    username='hashicorp' \
    rotation_period="6h" 
```
2. create a VaultDynamicSecret configured to read that static role:
```
---
apiVersion: secrets.hashicorp.com/v1alpha1
kind: VaultDynamicSecret
metadata:
  namespace: default
  name: example-dyn
spec:
  vaultAuthRef: example
  mount: ldap
  path: static-cred/hashicorp
  destination:
    create: true
    name: dynamic1
  rolloutRestartTargets:
        -  kind: "Deployment"
           name: "deployment-smb"
```
3. trigger a manual rotation to save time:
```
vault write -f ldap/rotate-role/hashicorp
```
4. see that no rollout was triggered, and the secret dynamic1 was not updated.

**Expected behavior**
the new password should be updated to the k8s secret "dynamic1" and the deployment should trigger a rollout.

**Environment**
* Kubernetes version: 
EKS 1.24
* vault-secrets-operator version:
Latest.

**Comment by benashz** [comment]
Thanks for reporting this issue @ninjadude333 . The VDS controller does not seem properly support the dynamic secret+static role use case. In order to add support we will mostly likely have to introduce some new configuration. We should be able to get this in for the upcoming RC release.

**Comment by ninjadude333** [comment]
great thanks

**Comment by ninjadude333** [comment]
any ETA ?

**Comment by benashz** [comment]
We're hoping to have this in for the upcoming GA release ~ June 11th or so.

**Comment by ninjadude333** [comment]
cool

**Comment by benashz** [comment]
Fixed in #239
